test/instances/eeth/EETHHyperdrive.t.sol

```diff
@@ -0,0 +1,618 @@
+        } else {
+            // Ensure that the amount of pooled ether stays the same.
+            assertEq(POOL.getTotalPooledEther(), totalBaseBefore);
+
+            // Ensure that the ETH balances were updated correctly.
+            assertEq(
+                address(hyperdrive).balance,
+                hyperdriveBalancesBefore.ETHBalance
+            );
+            assertEq(trader.balance, traderBalancesBefore.ETHBalance);
+
+            // Ensure that the EETH balances were updated correctly.
+            assertApproxEqAbs(
+                EETH.balanceOf(address(hyperdrive)),
+                hyperdriveBalancesBefore.baseBalance + amountPaid,
+                2
+            );
+            assertApproxEqAbs(
+                EETH.balanceOf(trader),
+                traderBalancesBefore.baseBalance - amountPaid,
+                2
+            );
+
+            // Ensure that the EETH shares were updated correctly.
+            uint256 expectedShares = amountPaid.mulDivDown(
+                totalSharesBefore,
+                totalBaseBefore
+            );
+            assertEq(EETH.totalShares(), totalSharesBefore);
+            assertApproxEqAbs(
+                EETH.shares(address(hyperdrive)),
+                hyperdriveBalancesBefore.sharesBalance + expectedShares,
+                2
+            );
+            assertApproxEqAbs(
+                EETH.shares(trader),
+                traderBalancesBefore.sharesBalance - expectedShares,
+                2
+            );
+        }
+    }
+
+    /// @dev Verifies that withdrawal accounting is correct when closing positions.
+    function verifyWithdrawal(
+        address trader,
+        uint256 baseProceeds,
+        bool asBase,
+        uint256 totalBaseBefore,
+        uint256 totalSharesBefore,
+        AccountBalances memory traderBalancesBefore,
+        AccountBalances memory hyperdriveBalancesBefore
+    ) internal view override {
+        // Base withdraws are not supported for this instance.
+        if (asBase) {
+            revert IHyperdrive.UnsupportedToken();
+        }
+
+        // Ensure that the total pooled ether and shares stays the same.
+        assertEq(POOL.getTotalPooledEther(), totalBaseBefore);
+        assertApproxEqAbs(EETH.totalShares(), totalSharesBefore, 1);
+
+        // Ensure that the ETH balances were updated correctly.
+        assertEq(
+            address(hyperdrive).balance,
+            hyperdriveBalancesBefore.ETHBalance
+        );
+        assertEq(trader.balance, traderBalancesBefore.ETHBalance);
+
+        // Ensure that the EETH balances were updated correctly.
+        assertApproxEqAbs(
+            EETH.balanceOf(address(hyperdrive)),
+            hyperdriveBalancesBefore.baseBalance - baseProceeds,
+            1e4
+        );
+        assertApproxEqAbs(
+            EETH.balanceOf(trader),
+            traderBalancesBefore.baseBalance + baseProceeds,
+            1e4
+        );
```

Comment:
These tolerances can probably be brought down when we update the conversions library